src/java/org/sogive/data/loader/ImportEditorialsDataTask.java

```diff
@@ -0,0 +1,58 @@
+package org.sogive.data.loader;
+
+import com.winterwell.utils.log.Log;
+import org.sogive.data.charity.NGO;
+
+import java.io.IOException;
+
+/**
+ * Imports editorials from a given published Google Doc url.
+ * @author anita
+ *
+ */
+public class ImportEditorialsDataTask {
+
+	private static final String TAG = ImportEditorialsDataTask.class.getSimpleName();
+
+	private volatile boolean running;
+
+	private final EditorialsFetcher editorialsFetcher;
+	private final DatabaseWriter database;
+
+	public ImportEditorialsDataTask(JsoupDocumentFetcher jsoupDocumentFetcher, DatabaseWriter database) {
+		editorialsFetcher = new EditorialsFetcher(jsoupDocumentFetcher);
+		this.database = database;
+	}
+
+	public synchronized void run(String publishedGoogleDocsUrl) {
+		running = true;
+		Editorials editorials;
```

Comment:
Wrap this in a try-catch-finally to ensure running=false gets reset at the end, even if there's an error